engine/src/player.ts

```diff
@@ -731,9 +733,10 @@ export default class Player extends EventEmitter {
     this.receiveTriggerIncome(Condition.Federation);
   }
 
-  factionReward(reward: Reward): Reward {
+  factionReward(reward: Reward, allowConversion = true): Reward {
```

Comment:
I'm really not a fan of carrying a parameter across multiple code lines: `loadEvents`, `loadEvent`, `gainRewards`... just for one specific faction, and their variant.

Maybe this should call `this.handlers.factionReward.call(this, reward)` so all the gleens-specific code can be moved to gleens.

Or maybe it's even possible to just add this handler to gleens:

```
handlers: {
  'gain-q': function(this: Player, count: number, source: ...) {
     if (!this.academy2) {
       // gain equivalent number of ores
       this.gainReward(new Reward(Resource.Ore, count), false, source);
       // remove the QIC
       this.gainReward(new Reward(Resource.Qic, count), true, source)
     }
  }
}
```

And remove `factionRewards` entirely

Reply:
Adding `factionReward`, separate of `handlers`, to `defaultBoard` seems to be the best option:

```
const defaultBoard = {
   ...,
   handlers: {},
   factionReward(this; Player, reward: Reward) {
      return reward
   }
}
```

this will prevent sideeffects that the `gain-q` handler would introduce

Reply:
problem is mainly knowing that it's the start of the game (i.e. round == 0) where there is no conversion. How would that work in your proposal?

Reply:
One possible solution:

```
    if (!skipIncome) {
      this.loadTechs(expansions);
      this.loadEvents(this.board.income);
      this.emit('boardIncomeLoaded'); // here
    }
```

and add a handler:

```
  handlers: {
    'boardIncomeLoaded': function(this: Player) {
       this.data.gainRewards([new Reward('q'), new Reward(-1, Resource.ore)])
    }
  }
```

Reply:
Another solution: add a paramater `source` to `factionReward`:

```
      this.data.gainRewards(
        rewards.map((rew) => this.factionReward(rew, source)), // here
        false,
        source
      );
```

And forward it to `this.board.factionReward(reward, source)`

Reply:
I like the idea with source :smile: